clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -520,14 +520,29 @@ void CGDebugInfo::CreateCompileUnit() {
   // the file name itself with no path information. This file name may have had
   // a relative path, so we look into the actual file entry for the main
   // file to determine the real absolute path for the file.
+  // An exception here is workflow when integration footer is involved: in that
+  // case driver passes an absolute path to the original user-provided source
+  // file, whilst main file corresponds to a temporary file generated by the
+  // compiler.
   std::string MainFileDir;
   if (const FileEntry *MainFile = SM.getFileEntryForID(SM.getMainFileID())) {
     MainFileDir = std::string(MainFile->getDir()->getName());
+    FileID MainFileID = SM.getMainFileID();
     if (!llvm::sys::path::is_absolute(MainFileName)) {
       llvm::SmallString<1024> MainFileDirSS(MainFileDir);
       llvm::sys::path::append(MainFileDirSS, MainFileName);
       MainFileName =
           std::string(llvm::sys::path::remove_leading_dotslash(MainFileDirSS));
+    } else {
```

Comment:
Instead of just 'else' here, I would think we would want to better check that this _IS_ an int-footer build.  Can we have this functionality opted-in somehow?

Reply:
I've introduced such option in e5b8bf8a7e8ef0942d17b83988b0e302cac7f478. The name is probably awful and it lacks testing, but what do you think about the overall approach? Did I get your comment correctly?